### ISSUE TITLE
Avoid requiring 'bs:disable-line for Roku_Ads() if library is imported

### ISSUE THREAD

**Issue by iObject** [rokucommunity/brighterscript#179]
We should avoid requiring 'bs:disable-line for Roku_Ads() if library is imported.

**Comment by TwitchBronBron** [comment]
I started investigating this like 6 months ago, but kept getting pulled into other things. We still need to add support for detecting the library import in the manifest, and then adding those functions into the global function scope if imported. 

@iObject is this something you would have any interest in implementing? We can have a quick chat on Slack if you want to work out the implementation details. I don't expect it to be all that complicated.

**Comment by TwitchBronBron** [comment]
I did a some on-device investigation, just to get a broader understanding of the edge cases for this feature:

1. in `source` scope, the `Library "Roku_Ads.brs"` statement can appear in any file. It does not need to be in the same file as the call to `Roku_Ads()`, and does not need to be in the file with the `main()` function.
2. in a `Task` component scope, the `Library "Roku_Ads.brs"` statement can appear in any of the referenced script files. it does not need to be in the same file as the call to `Roku_Ads()`. 
2. Declaring the `Library` statement in `source` scope, but calling `Roku_Ads` in a component scope results in `function not found` error
3. Declaring the `Library` statement in a component scope, and in the same file calling `Roku_Ads()` fails with `[RAF.err] RAF call in wrong environment (RENDER thread?). RAF only works from task or main() thread`


So, to summarize: 
1. the `Library "Roku_Ads.brs"` statement must be declared in a script in the same scope as the call to `Roku_Ads()`
2. Component scopes will crash when calling `Roku_Ads()` even though they support the `Library "Roku_Ads.brs"` statement (it's useless)

**Comment by elliot-nelson** [comment]
Copying some ideas from discussion in Slack:

In your own Roku project, you could create a `.d.bs` file that describes the function you are using, for example:

```brighterscript
' src/source/Roku_Ads.d.bs

function Roku_Ads()
end function
```

Now you can call this `Roku_Ads` function from your main thread:

```brighterscript
Library "Roku_Ads.brs"

sub main()
  Roku_Ads()
end sub
```

You can also call it from a task, or component, by importing the `.d.bs` file:

```brighterscript
import "../source/Roku_Ads.d.bs"

sub init()
  Roku_Ads()
end sub
```

This allows unblocking any libraries used in your own project.

Eventually, we could have a library of user-supplied `.d.bs` files with function definitions for commonly used libraries, and you could just pull from there (like the `@types/` project for typescript types).

**Comment by rrirower** [comment]
Is there any way to import Roku_Ads in a component script?  I've tried the above workaround, but, it does not work. I can include the Library in a Task node, but, if I try to use it in a component script it fails with:

`Could not load source 'roku_ads_lib:/libsource/Roku_Ads_Beacons.brs'.`
